src/com/google/javascript/jscomp/PeepholeFoldConstants.java

```diff
@@ -1064,15 +1064,22 @@ static TernaryValue evaluateComparison(int op, Node left, Node right, boolean us
 
       case Token.STRING:
         if (right.isString()) {
-          // Only eval if they are the same type
+          return compareStrings(op, left, right);
+        } else {
+          // See http://www.ecma-international.org/ecma-262/6.0/#sec-abstract-relational-comparison
           switch (op) {
-            case Token.SHEQ:
+            case Token.GE:
+            case Token.LE:
+            case Token.GT:
+            case Token.LT:
+              return compareAsNumbers(op, left, right, useTypes);
```

Comment:
I dislike this pattern in that comparing a number to (something else) doesn't necessarily mean you should compare them as numbers.

Consider, for example,
`'1' < '0' + '2'`

If we (correctly) improved our implementation of Node.getNumberValue to be able to handle string addition, it would cause this code to become incorrect.